File: fall_2020/coding_meetup_01/chocolate_division/chocolate_division.cpp
/// @author Benjamin Alcocer

#include <iostream>
#include <algorithm>

using namespace std;

int main()
{
    long long int n, m, d;

    // https://www.geeksforgeeks.org/stdgcd-c-inbuilt-function-finding-gcd/
    // Since it is not C++17
    // Otherwise, C++17: https://en.cppreference.com/w/cpp/numeric/gcd
    cin >> n >> m;
    d = __gcd(n, m);
    n /= d;
    m /= d;
    n % 2 == 1 && m % 2 == 1 ? cout << d : cout << '0';

    return 0;
}

File: 2020_fall/coding_meetup_01/chocolate_division/chocolate_division.py
# @author Benjamin Alcocer

import math

array = input().replace('\n', '').split()
n = int(array[0])
m = int(array[1])
d = math.gcd(n,m)
n = n // d
m = m // d
print(d if n % 2 == 1 and m % 2 == 1 else 0)
